drivers/sensor/bme280/bme280.c

```diff
@@ -584,6 +599,59 @@ int bme280_init(const struct device *dev)
 	return rc;
 }
 
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+int bme280_pm_ctrl(const struct device *dev, uint32_t ctrl_command,
+				void *context, device_pm_cb cb, void *arg)
+{
+	struct bme280_data *data = to_data(dev);
+
+	int ret = 0;
+
+	/* Set power state */
+	if (ctrl_command == DEVICE_PM_SET_POWER_STATE) {
+		uint32_t new_pm_state = *((const uint32_t *)context);
+
+		if (new_pm_state != data->pm_state) {
+
+			/* Switching from OFF to any */
+			if (data->pm_state == DEVICE_PM_OFF_STATE) {
+
+				/* Re-initialize the chip */
+				ret = bme280_chip_init(dev);
+			}
+			/* Switching to OFF from any */
+			else if (new_pm_state == DEVICE_PM_OFF_STATE) {
+
+				/* Put the chip into sleep mode */
+				ret = bme280_reg_write(dev,
+					BME280_REG_CTRL_MEAS,
+					BME280_CTRL_MEAS_OFF_VAL);
+
+				if (ret < 0)
+					LOG_DBG("CTRL_MEAS write failed: %d",
+						ret);
+			}
+
+			/* Store the new state */
+			if (!ret)
+				data->pm_state = new_pm_state;
+		}
+	}
+	/* Get power state */
```

Comment:
Why not implementing DEVICE_PM_GET_POWER_STATE?

Reply:
It is implemented in the following else statement, lines 641 - 644.

Reply:
> It is implemented in the following else statement, lines 641 - 644.

Sorry, I missed it!